mne/io/eeglab/eeglab.py

```diff
@@ -260,6 +327,247 @@ def read_epochs_eeglab(input_fname, events=None, event_id=None, montage=None,
+
+            elif curr == 'event':
+                temp = _hlGroup_2_bunch_list(orig, in_hdf[curr], curr,
+                                             indent + indent_incr)
+
+                # TO DO add tests to know when to add
+                # these (& other) string fields
+                str_conversion_fields = ('type', 'usertags')
+                temp = _bunch_str_conversions(temp, str_conversion_fields)
+                temp = np.asarray(temp)
+
+            elif curr == 'epoch':
+                temp = _hlGroup_2_bunch_list(orig, in_hdf[curr],
+                                             curr_name, indent + indent_incr)
+
+                deref_fields = ('eventtype', 'eventlatency', 'eventurevent',
+                                'eventduration', 'eventvalue')
+                temp = _bunch_derefs(orig, temp, deref_fields)
+
+                for curr_elem in temp:
+                    eventtype_str = [''.join([chr(x) for x in c_evt])
+                                     for c_evt in curr_elem.eventtype]
+                    curr_elem.eventtype = np.asarray(eventtype_str)
+
+            else:
+                temp = hdf_2_dict(orig, in_hdf[curr],
+                                  curr_name, indent + indent_incr)
+            out_dict[curr] = temp
+
+        else:
+            sys.exit("Unknown type")
+
+    return out_dict
+
+
+def _hlGroup_2_bunch_list(orig, in_hlGroup, tuple_name, indent):
+    """Returns list of Bunch objects - A Bunch object is a
+       dictionary-like object that exposes its keys as attributes.
+       ASSUMES: The group consists solely of arrays of HDF5 obj refs,
+       and that these refs all reference 2D numpy arrays that need to
+       be flattened to either 1D arrays or Scalars"""
+
+    import h5py
+
+    try:
+        # h5_values gives dict of 1D arrays of HDF obj references
+        h5_values = {ct: in_hlGroup[ct].value.flatten() for ct in in_hlGroup}
+
+        # derefs dereferences HDF obj references and cnverts arrays with
+        # shapes = (1,) to scalars. Returns adict mapping keys to lists
+        # of arrays and scalars
+        derefs = {x: [orig[y].value.flatten()
+                      if orig[y].value.flatten().shape != (1,)
+                      else orig[y].value.flatten()[0]
+                      for y in h5_values[x]]
+                  if isinstance(h5_values[x], np.ndarray) and
+                  isinstance(h5_values[x][0], h5py.Reference)
+                  else h5_values[x]
+                  for x in h5_values}
+
+        for ct in in_hlGroup:
+            msg = indent + "Converting " + tuple_name + '_' + ct
+            logger.debug(msg)
+
+    except IOError:
+        derefs = {ct: [None] for ct in in_hlGroup}
+        warn("Couldn't read", tuple_name, ". Assuming empty")
+
+    sz = len(derefs[list(derefs.keys())[0]])
+    bch_list = [Bunch(**{key: derefs[key][x] for key in derefs})
+                for x in range(sz)]
+    return bch_list
```

Comment:
This code looks like it was all taken from somewhere else, basically to convert modern `.mat` files to the same format legacy `.mat` files would yield when used with `scipy.io.loadmat`, correct? If so, put all of these functions in `mne/externals/something.py`

Reply:
... this comment is the one I reference above